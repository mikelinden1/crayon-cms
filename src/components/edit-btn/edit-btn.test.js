import EditBtn from './edit-btn';
import sinon from 'sinon';

const props = {
    item: {
        name: 'test'
    },
    config: {
        capabilities: {
            editable: true
        }
    },
     actions: {
        editItem: () => {}
     }
};

it('EditBtn matches snapshot', () => {
    const wrapper = shallow(<EditBtn {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it('EditBtn doesn\'t display if editable not set', () => {
    const propsNoEdit = {
        ...props,
        config: {
            capabilities: {
                editable: false
            }
        }
    };

    const wrapper = shallow(<EditBtn {...propsNoEdit} />);
    expect(wrapper.type()).toEqual(null)
});

it('EditBtn calls editItem and sends the item', () => {
    const spy = sinon.spy();
    const propsWithSpy = {
        ...props,
        actions: {
            editItem: spy
        }
    };

    const wrapper = mount(<EditBtn {...propsWithSpy} />);
    wrapper.find('Button').first().simulate('click');

    expect(spy.calledOnce).toBeTruthy();
    expect(spy.getCall(0).args[0]).toEqual(propsWithSpy.item);
});