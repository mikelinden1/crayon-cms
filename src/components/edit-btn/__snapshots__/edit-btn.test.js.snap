// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBtn matches snapshot 1`] = `
<Button
  color="primary"
  onClick={[Function]}
  tag="button"
>
  Edit
</Button>
`;
